### PR TITLE
#160370625 Write test for edit comment functionality

### PR DESCRIPTION
**Pull request by hariclerry**
#### What does this PR do?
This PR adds the test to cater for edit comment feature.

#### Description of Task to be completed?
Write test for edit comment feature, the following test files contain the written test;      

1.        src/components/RequestsModal/CommentBox/__tests__/CommentBox.test.js
2.         src/components/RequestsModal/UserComments/__tests__/UserComments.test.js
3.         src/redux/actionCreator/__tests__/commentsActions.test.js
4.         src/redux/middleware/__tests__/commentsSaga.test.js
5.         src/redux/reducers/__tests__/comments.test.js
6.         src/services/__tests__/CommentsAPI.test.js 

#### How should this be manually tested?
- Clone the repository
- Switch to the repo directory - `cd travel_tool_front`
- Run `make test` to run all the test files
- On the console, you should be able to see the test coverage displaying test covered 
#### Any background context you want to provide?
None

#### What are the relevant pivotal tracker stories?
[#160370625](https://www.pivotaltracker.com/story/show/160370625)

#### Screenshots (if appropriate)
None

#### Questions:
None

### CODE REVIEW

**Comment by Nedson202** [comment]
Hey @hariclerry, on the `How should this be manually tested` section, is the `make start` section really important since this PR aims solely at testing the already implemented feature?

**Comment by hariclerry** [comment]
@Nedson202, thanks for your feedback, I have made the requested changes.

File: src/components/RequestsModal/CommentBox/__tests__/CommentBox.test.js [hunk 1]
```diff
@@ -6,6 +6,8 @@ describe('Render CommentBox component', () => {
   
   const props = {
     createComment: jest.fn(),
+    handleEditComment: jest.fn(),
+    editComment:  jest.fn(),
     handleSubmit:()=>{},
   };
   const e = {
```
Comment (clintfidel) [inline src/components/RequestsModal/CommentBox/__tests__/CommentBox.test.js]: Can this variable name be more explicit as I don't know what `e` means

File: src/components/RequestsModal/UserComments/__tests__/UserComments.test.js [hunk 2]
```diff
@@ -29,9 +32,37 @@ describe('UserComments component', () => {
...
+  });
+
+  it('should call handleCancel', () => {
+    const wrapper = shallow(<UserComments {...props} />);
+    const spy = jest.spyOn(wrapper.instance(), 'handleCancelClick');
+    const fakeEvent = { preventDefault: () => {} };
+    wrapper.setState({
+      commentToEdit: 'Can you clarify why'
+    });
+    let cancelButton = wrapper.find('#cancel-button');
```
Comment (Nedson202) [inline src/components/RequestsModal/UserComments/__tests__/UserComments.test.js]: I see that you are declaring variables using the `let` keyword. If the variables are not reassigned new values, I think we should stick with `const`.

Comment (Nedson202) [reply src/components/RequestsModal/UserComments/__tests__/UserComments.test.js]: There are some other instances of this. Please look into it.

**Comment by hariclerry** [comment]
Hey @Nedson202 and @clintfidel, the requested changes have been implemented. Thanks.

**Pull request merged**